sanzu/src/client_x11.rs

```diff
@@ -758,10 +715,12 @@ impl Client for ClientInfo {
 
         self.conn.flush().context("Error in x11rb flush")?;
         self.conn
-            .free_cursor(cid)
+            .free_cursor(cursor)
```

Comment:
Small suggestions in case you are interested:

In my PR, I made this code use `GcontextWrapper` and `PixmapWrapper` to automatically free things on `Drop`. There are also `CursorWrapper` and `render::PictureWrapper`. Each of those have a `for_foo()` function to create one for an already existing resource or a `create_foo` function that also deals with `generate_id()` etc. I didn't use `create_gcontext()` in my PR since you already have your own `create_gc` helper function.

Random link to the docs: https://docs.rs/x11rb/latest/x11rb/protocol/render/struct.PictureWrapper.html

Reply:
Ok: I didn't checked the doc for those ones. Shame on me.
I made #173 to correct this.

Tell me if I am wrong but, the CursorWrapper `create_cursor` takes a `pixmap` as source, so as I want `cursor` here, I can only use the `for_cursor` in this case as you advised, right?
I mean I didn't see like the others a create function which can create a cursor for a picture + hot spot. Did I miss something?

Thanks again for your review @psychon !

Reply:
> I can only use the for_cursor in this case as you advised, right?

Yeah, you are right.

> Did I miss something?

Nope, you did not.

`render::create_cursor` comes from the render extension, but a cursor is in `xproto` and hence `CursorWrapper` is in `xproto`. Such a cross-extension case doesn't work with the mechanism we use for generating these wrapper structs. That's why there isn't a convenience helper for that case.